src/webrtc/remoteCallFeed.ts

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "../logger";
+import { CallEvent, CallState, MatrixCall } from "./call";
+import { SDPStreamMetadataObject, SDPStreamMetadataPurpose } from "./callEventTypes";
+import { CallFeed, CallFeedEvent, ICallFeedOpts } from "./callFeed";
+import { RemoteCallTrack } from "./remoteCallTrack";
+
+export interface RemoteCallFeedOpts extends ICallFeedOpts {
+    streamId: string;
+    metadata?: SDPStreamMetadataObject;
+    call: MatrixCall;
+
+    /**
+     * @deprecated addTransceiver() should be used instead
+     */
+    stream?: MediaStream;
+}
+
+export class RemoteCallFeed extends CallFeed {
+    private _connected = false;
```

Comment:
`RemoteCallFeed` holds as well the info of the `CallState`? What is with `Screen Share`? We cover this as well in a extra `RemoteCallFeed` with a extra `CallState`?

Reply:
Sorry, I am not sure I understand... `CallState` has nothing to do with screen-sharing - the possible values are connected, connecting, ended...

Reply:
Maybe my question have been misleading.  I explain in some detail.

Why i have used the Screen Share example:
If a user receives a ScreenShare, we also cover this MediaStream as a `RemoteCallFeed`. Receiving Screen Share is an example that's an user receiving two media streams from the same remote user as well in the non SFU case. 

To the property itself: 
Your property `connected` maps two states into one: 'CallState' and basically `remote Track received (yes|no)`.

I was confused that the connection status is shown in the remote call feed because:
In the case of non SFU the change of `CallState` is the same for all `RemoteCallFeeds` from the same Remote User and in case of SFU the change of `CallState` is the same for all `RemoteCallFeeds`. 

With the cover of the state: `remote Track received (yes|no)`, you only cover the condition that the metadata event arrives before the remote media stream. This is the only case in which a `RemoteCallFeed` without a remote MediaStream without tracks exists.

In summary, `connected ` essentially maps the ICE connection state of the underlying WebRTC connection. Putting this state in the feeds is a fundamental design decision, so I wanted to question that.

Reply:
I am still having a bit of trouble but do I understand you correctly you're wondering why the `connected` property is on all `CallFeed`s?

The answer there is that I wanted to hide all the underlying connection logic from the js-sdk consumer. Basically, the consumer only needs to create a group call and not care if it's full-mesh or SFU as it can read the `connected` property from the individual feeds instead of getting the individual calls and figuring out if they're connected or not. This doesn't feel right as it makes the consumer have to care about the full-mesh vs SFU difference and other stuff

Does that make sense?

Reply:
Exactly, I'm asking about the API design. And yes, your intention makes absolutely sense to me. It's also the software design how Element Call works. If a connection is aborted, we will see it in the media stream tile. 

It's not a small design decision and it affects how the SDK is used. If this design decision is general consensus then that's ok and we can close this as solved.

Reply:
I believe it is, since it passed review before, shout if you disagree 😅